core/src/main/scala/latis/dataset/GranuleAppendDataset.scala

```
@@ -68,14 +77,15 @@ class GranuleAppendDataset private (
    */
   def samples: Stream[IO, Sample] =
     Stream.eval(
-      granuleList                          //start with granule list dataset
-        .withOperations(listOps)           //add operations to granule list
-        .samples                           //get samples, one for each granule
-        .map(granuleToDataset)             //convert each sample to a dataset
-        .compile.toList                    //get the list of datasets from stream (in IO)
-        .map(makeDataset)                  //combine the granules into a single dataset
-        .map(_.withOperations(operations)) //add operations to dataset
-    ).flatMap(_.samples)                   //eval back into stream then get samples
+      granuleList                           //start with granule list dataset
+        .withOperations(listOps)            //add operations to granule list
+        .samples                            //get samples, one for each granule
+        .map(s => Try(granuleToDataset(s))) //convert each sample to a dataset
```

Does `granuleToDataset` throw exceptions? I think it would be better to use `Either.catchNonFatal` or `Either.catchOnly`.

---

Is your concern that we might swallow fatal errors?

I suppose we should catch only `LatisException`s (i.e. ones we should know something about) and log a warning.

---

No, it's just that `Try` doesn't really add anything over `Either` when you're using `cats`.